browser-sign-in/App.js

```diff
@@ -0,0 +1,220 @@
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+ 
+import React from 'react';
+
+import { Button, StyleSheet, Text, View } from 'react-native';
+import { 
+  createConfig, 
+  signIn, 
+  signOut, 
+  getAccessToken, 
+  isAuthenticated,
+  getUser,
+  getUserFromIdToken,
+  EventEmitter
+ } from '@okta/okta-react-native';
+ import configFile from './samples.config';
+
+export default class App extends React.Component {
+  constructor() {
+    super()
+    this.state = {
+      authenticated: false,
+      context: null
+    }
+    this.checkAuthentication = this.checkAuthentication.bind(this);
+    this.getMessages = this.getMessages.bind(this);
+  }
+  
+  async componentDidMount() {
+    let that = this;
+    EventEmitter.addListener('signInSuccess', function(e: Event) {
+      that.setState({ authenticated: true });
+      that.setContext('Logged in!');
+    });
+    EventEmitter.addListener('signOutSuccess', function(e: Event) {
+      that.setState({ authenticated: false });
+      that.setContext('Logged out!');
+    });
+    EventEmitter.addListener('onError', function(e: Event) {
+      console.log(e);
+    });
+    EventEmitter.addListener('onCancelled', function(e: Event) {
+      console.warn(e);
+    });
+    await createConfig({
+      clientId: configFile.oidc.clientId,
+      redirectUri: configFile.oidc.redirectUri,
+      endSessionRedirectUri: configFile.oidc.endSessionRedirectUri,
+      discoveryUri: configFile.oidc.discoveryUri,
+      scopes: configFile.oidc.scopes,
+      requireHardwareBackedKeyStore: configFile.oidc.requireHardwareBackedKeyStore
+    });
+    this.checkAuthentication();
+  }
+  
+  componentWillUnmount() {
+    EventEmitter.removeAllListeners();
+  }
+
+  async componentDidUpdate() {
+    this.checkAuthentication();
+  }
+
+  async checkAuthentication() {
+    const result = await isAuthenticated();
+    if (result.authenticated !== this.state.authenticated) {
+      this.setState({ authenticated: result.authenticated });
+    }
+  }
+
+  async login() {
+    signIn();
+  }
+
+  async logout() {
+    signOut();
+  }
+
+  async getUserIdToken() {
+    let user = await getUserFromIdToken();
+    this.setContext(`
+      User Profile:
+      ${JSON.stringify(user, null, 4)}
+    `);
+  }
+  
+  async getMyUser() {
+    let user = await getUser();
+    this.setContext(`
+      User Profile:
+      ${user}
+    `);
+  }
+
+  async getMessages() {
```

Comment:
I think this function doesn't make for the RN sample app. I'm as RN developer doesn't want to download server and run it on my machine. I just want to play with my existing preview org.
We can probably redo `getMyUser` function and show how to use access token to get information about the user. So instead of bridging into native code just show how to use access token in js part

Reply:
Added another method to fetch user info with access token